### PR TITLE
Reduce LOC in security context tests

### PR DESCRIPTION
**Pull request by pmorie**
Make this test a little easier to read w/ extract method refactor

@pweil- @erictune

### CODE REVIEW

File: pkg/securitycontext/util_test.go [hunk 1]
```diff
@@ -84,6 +84,10 @@ func compareContexts(name string, ex, ac *api.SELinuxOptions, t *testing.T) {
 	}
 }
 
+func container(*int64) api.Container {
```
Comment (pweil-) [inline pkg/securitycontext/util_test.go]: nit: I would make this `containerWithUser` and give a name to the method parameter for clarity.

**Comment by pweil-** [comment]
minor nit and LGTM

**Comment by pmorie** [comment]
@pweil- feedback addressed

**Comment by erictune** [comment]
LGTM

**Comment by pweil-** [comment]
LGTM

**Comment by erictune** [comment]
@k8s-bot please e2e test this flake #22719

**Comment by pmorie** [comment]
@k8s-bot please e2e test this flake #22719

**Comment by pmorie** [comment]
@k8s-bot please e2e test this flake #22719

**Comment by pmorie** [comment]
@k8s-bot please e2e test this flake #22719

**Comment by pmorie** [comment]
@k8s-bot please e2e test this flake #22719

**Comment by pmorie** [comment]
@k8s-bot please e2e test this flake #22719

**Comment by pmorie** [comment]
@k8s-bot please e2e test this flake #22719

**Comment by pmorie** [comment]
@k8s-bot please e2e test this flake #22719

**Comment by pmorie** [comment]
@k8s-bot please e2e test this flake #22719

**Comment by pmorie** [comment]
@k8s-oncall May I enter a plea for a manual merge?  I've been fighting flakes on this for a month, they are in no way related to the lines of unit test code this PR changes.  At this point I either need this to be merged or I'll just close it.

**Comment by janetkuo** [comment]
@k8s-bot e2e test this issue: #22719

**Comment by a-robinson** [comment]
Sigh, I was just about to merge and then it gets retriggered... I feel your pain, @pmorie

**Pull request merged**